src/answers-headless.ts

```diff
@@ -27,108 +27,247 @@ import answersUtilities from './answers-utilities';
+   */
   setAlternativeVerticals(alternativeVerticals: VerticalResults[]): void {
     this.stateManager.dispatchEvent('vertical/setAlternativeVerticals', alternativeVerticals);
   }
 
+  /**
+   * Sets {@link VerticalSearchState.sortBys} to the specified sortBys.
+   *
+   * @param sortBys - The new sortBys to update to
+   */
   setSortBys(sortBys: SortBy[]): void {
     this.stateManager.dispatchEvent('vertical/setSortBys', sortBys);
   }
 
+  /**
+   * Sets {@link MetaState.context} to the specified context.
+   *
+   * @param context - The new context to update to
+   */
   setContext(context: Context): void {
     this.stateManager.dispatchEvent('meta/setContext', context);
   }
 
+  /**
+   * Sets {@link MetaState.referrerPageUrl} to the specified URL.
+   *
+   * @param referrerPageUrl - The new referring page URL to update to
+   */
   setReferrerPageUrl(referrerPageUrl: string): void {
     this.stateManager.dispatchEvent('meta/setReferrerPageUrl', referrerPageUrl);
   }
 
+  /**
+   * Sets {@link LocationState.userLocation} to the specified latitude and
+   * longitude.
+   *
+   * @param latLong - The new user location to update to
+   */
   setUserLocation(latLong: LatLong): void {
     this.stateManager.dispatchEvent('location/setUserLocation', latLong);
   }
 
+  /**
+   * Sets the {@link State} to the specified state.
+   *
+   * @param state - The new state to update to
+   */
   setState(state: State): void {
     this.stateManager.dispatchEvent('set-state', state);
   }
 
+  /**
+   * Sets {@link UniversalSearchState.restrictVerticals} to the specified vertical
+   * keys.
+   *
+   * @param restrictVerticals - The new verticals to restrict a universal search
+   */
   setRestrictVerticals(restrictVerticals: string[]): void {
     this.stateManager.dispatchEvent('universal/setRestrictVerticals', restrictVerticals);
   }
 
+  /**
+   * Gets the current state of the AnswersHeadless instance.
+   */
   get state(): State {
     return this.stateManager.getState();
   }
 
+  /**
+   * Adds a listener for a specific state value of type T.
+   *
+   * @param listener - The listener to add
+   * @returns The function for removing the added listener
+   */
   addListener<T>(listener: StateListener<T>): Unsubscribe {
     return this.stateManager.addListener<T>(listener);
   }
 
+  /**
+   * Uses {@link AnswersCore} to submit a question to the Answers API with the
+   * specified request data.
+   *
+   * @param request - The data for the network request
+   * @returns A Promise of a {@link QuestionSubmissionResponse} from the Answers API
+   */
   async submitQuestion(request: QuestionSubmissionRequest): Promise<QuestionSubmissionResponse> {
     return this.core.submitQuestion(request);
   }
 
-  async executeUniversalQuery(): Promise<UniversalSearchResponse | undefined> {
+  /**
+   * Uses {@link AnswersCore} to perform an Answers search across all verticals with
```

Comment:
Again, I don't think we need to indicate that `AnswersCore` is being used here. How about something like: "Performs an answers search across all verticals with relevant parts of the state used as input to the search"?

Reply:
Updated this and the others similarly